### PR TITLE
README: overhaul file structure

### LINKED ISSUE #777: About text should be updated to remove reference to usage as a wrapper
Currently the "About" text says:

```
About

Lab wraps Git or Hub, making it simple to clone, fork, and interact with repositories on GitLab 
```

From my understanding the project has moved away from recommending `lab` as a wrapper around `git`.

This is the "About" text seen when going to: https://github.com/zaquestion/lab

### PR DESCRIPTION
**Pull request by bmeneg**
In order to make the README information more clear this patch overhaul
its structure and update its content to match the current state of the
tool.

Signed-off-by: Bruno Meneguele <bmeneg@redhat.com>

Fix #777 .
This PR semantically depends on PR #809 being merged, otherwise some recommendations won't work as expected.

### CODE REVIEW

**Comment by bmeneg** [comment]
@zaquestion once PR #809 and this one here are merged, could you update the project "About" text dropping the "Lab wraps Git and Hub" part?

File: README.md
Comment (zampierilucas) [inline README.md:126]: Other parts of the README, such as installation, used triple `#` to sub-items, IMHO using tree `#` is more readable, as is makes and clearer distinction for topics and subtopics.
so 
```suggestion
### CI environment variables
```

File: README.md
Comment (zampierilucas) [inline README.md:132]: Same as above:
```suggestion
### Local environment variables
```

File: README.md
Comment (zampierilucas) [inline README.md:139]: Same as above:
```suggestion
### Configuration file
```

File: README.md
Comment (zampierilucas) [inline README.md:178]: Same as above:
```suggestion
### Fresh start
```

**Review by zampierilucas — APPROVED** [review]

File: README.md
Comment (zaquestion) [inline README.md:65]: It looks like lab still depends on `git` in `internal/git`? Tho I think this will be true once the `go-git` stuff goes in.

**Comment by zaquestion** [comment]
Might be worth adding a table of contents while you're at it, Readme has gotta a lot longer since the early days

File: README.md
Comment (bmeneg) [reply README.md:65]: Yes. But that's also why I used the "compilation-time" and "runtime" words there: the _git_ binary is only required during runtime, but not compilation.

**Comment by bmeneg** [comment]
Oh, sure, I'll add the toc :)

**Comment by bmeneg** [comment]
@zaquestion I just realized that GitHub itself has a table of contents in the left of readme's filename:
image

It was released [mid last year](https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/). With that I think we don't really need to have a specific toc in our doc.

File: README.md
Comment (bmeneg) [reply README.md]: Fixed.

File: README.md
Comment (bmeneg) [reply README.md]: I changed the wording a bit. Should be clearer now.

**Comment by bmeneg** [comment]
Lol.. I'm pretty sure a README file change don't affect a build testing:
```
409 {message: {base: [PG::QueryCanceled: ERROR:  canceling statement due to statement timeout
```
And I'm almost sure it's not client's problem (error 40x) .. hahaha.

**Comment by zampierilucas** [comment]
> @zaquestion I just realized that GitHub itself has a table of contents in the left of readme's filename: image
> 
> It was released [mid last year](https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/). With that I think we don't really need to have a specific toc in our doc.

Does GitLab support that? If not, It's not worth the work IMO, as we are moving soon™ to GitLab.

**Comment by bmeneg** [comment]
GitLab requires an explicit toc with `[[_TOC_]]`.
Since it's different for github we can change it once we migrate to GitLab.

File: README.md
Comment (prarit) [inline README.md:32]: _lab_ interacts

File: README.md
Comment (prarit) [inline README.md:38]: The development team has focused on keeping _lab_ a simple and intuitive command line interface for commands provided in the GitLab API (link?) .  _lab_'s aim is to provide GitLab users an experience similar to the GitLab webUI with respect to errors and messages.

File: README.md
Comment (prarit) [inline README.md:66]: In compilation-time, _lab_ depends only on other Go external modules, defined at `go.mod`.  At runtime, _git_ is required as many _git_ commands are used by lab to retrieve local repository information.

File: README.md
Comment (prarit) [inline README.md:131]: I've thought about this before and maybe it's time to remove this text?  It's only really useful in our CI environment and isn't something anyone else other than us is going to use.

File: README.md
Comment (prarit) [inline README.md:222]: We welcome all contributors and their contributions to lab!  See the [contribution guide](CONTRIBUTING.md).

@bmeneg, yeah it's silly :) but I do like to encourage others to contribute!

File: README.md
Comment (prarit) [inline README.md:224]: Should we continue to mention Glab?  I think this section can also be dropped.

**Review by prarit — CHANGES_REQUESTED** [review]

File: README.md
Comment (bmeneg) [reply README.md:131]: Not sure about that. I remember seeing someone in some issue mentioning they were using lab from within a CI env.

File: README.md
Comment (bmeneg) [reply README.md:38]: Yes, better :))

File: README.md
Comment (bmeneg) [reply README.md:66]: Thanks!

File: README.md
Comment (bmeneg) [reply README.md:222]: :D that's a good point

File: README.md
Comment (bmeneg) [reply README.md:224]: Yes, I thought about that too and was about to propose its removal. I don't really think this paragraph really helps one to understand what's the difference between the tool. And it refers to `hub` which is an already deprecated tool.
I'll remove it.

File: README.md
Comment (bmeneg) [reply README.md]: Whoops.

File: README.md
Comment (zaquestion) [reply README.md]: heh, I think I wasn't reading thoroughly enough, my bad, I completely overlooked the language about runtime and compile time here 😅

File: README.md
Comment (zaquestion) [inline README.md:162]: hmmm, I'll leave the choice to y'all, but personally am a bit hesitant to fully remove this section. Mostly because it was done out of solidarity for each others tool. They added a similar section in their readme to callout lab https://github.com/profclems/glab#what-about-lab

**Comment by zaquestion** [comment]
> @zaquestion once PR https://github.com/zaquestion/lab/pull/809 and this one here are merged, could you update the project "About" text dropping the "Lab wraps Git and Hub" part? I don't have permission to do so.

Can do, feel free to email me if I forget / fail to notice this merge.

File: README.md
Comment (bmeneg) [reply README.md:220]: @prarit , based on Zaq's comment below I let this section and even enhanced it to get closer to what is said about _lab_ in _glab_'s repo.

**Comment by bmeneg** [comment]
> > @zaquestion once PR #809 and this one here are merged, could you update the project "About" text dropping the "Lab wraps Git and Hub" part? I don't have permission to do so.
> 
> Can do, feel free to email me if I forget / fail to notice this merge.

:-) ack! Will do.

File: README.md
Comment (prarit) [reply README.md:127]: Good point.  But I still think this section needs work.  Right now when I read this section I come away with the idea that CI_* variables are the feature I'm supposed to use.  Perhaps reorganizing the subsections here so that the README contains Fresh start, Configuration file, then local, and then the CI variables would be better for new users?

File: README.md
Comment (prarit) [reply README.md:162]: Oh cool! :)  I had assumed badly that it was out of date :)

File: README.md
Comment (bmeneg) [reply README.md:127]: Ok, it makes sense indeed. Let me reorder them.

Comment (bmeneg) [reply README.md:123]: @prarit done. I slightly changed the _fresh start_ subsection.

Comment (bmeneg) [reply README.md:123]: @prarit any feedback on this?

**Review by prarit — APPROVED** [review]
LGTM!

**Comment by bmeneg** [comment]
@zaquestion, just fyi.. you can now update project's description. I'm hitting the merge button right now.

**Pull request merged**
